OpenSim/Examples/DataTable/example2.cpp

```diff
@@ -0,0 +1,82 @@
+/* -------------------------------------------------------------------------- *
+ *                            OpenSim:  example2.cpp                          *
```

Comment:
I don't see the reason for a separate example2 file. Seems like most of example1 is present in 2. It might be more informative and easier to maintain one example with different stages, like:

 ```C++
//Part 1: Populate a DataTable
...
//Part 2: Attempt to Convert a DataTable to a TimeSeriesTable
...
//Part 3: Update a DataTable to conform and convert to a TimeSeriesTable
...
```